src/d3a/models/strategy/storage.py

```diff
@@ -263,9 +279,13 @@ def buy_energy(self, market):
             if offer.seller == self.owner.name:
                 # Don't buy our own offer
                 continue
+            if "IAA" in offer.seller \
```

Comment:
This is OK for now, but I would leave a TODO to find a better way to distinguish between inter area agents and devices. This is a pending issue in other places as well, which we have not dealt with yet.